src/status_im/ui/screens/browser/permissions/views.cljs

```diff
@@ -0,0 +1,82 @@
+(ns status-im.ui.screens.browser.permissions.views
+  (:require-macros [status-im.utils.views :as views])
+  (:require [status-im.ui.components.animation :as anim]
+            [status-im.ui.components.react :as react]
+            [status-im.ui.screens.browser.styles :as styles]
+            [re-frame.core :as re-frame]
+            [status-im.i18n :as i18n]
+            [status-im.ui.components.common.common :as components.common]
+            [status-im.ui.components.icons.vector-icons :as icons]
+            [status-im.ui.components.colors :as colors]
+            [reagent.core :as reagent]
+            [status-im.models.browser :as model]
+            [status-im.ui.components.chat-icon.screen :as chat-icon.screen]))
+
+(views/defview permissions-panel [{:keys [dapp? name] :as browser} {:keys [requested-permission params]}]
+  (views/letsubs [dapp [:get-dapp-by-name name]
+                  bottom-anim-value (anim/create-value -354)
+                  alpha-value       (anim/create-value 0)
+                  hide-panel        #(anim/start
+                                      (anim/parallel
+                                       [(anim/spring bottom-anim-value {:toValue -354})
+                                        (anim/timing alpha-value {:toValue  0
+                                                                  :duration 500})]))]
+    {:component-did-mount #(anim/start
+                            (anim/parallel
+                             [(anim/spring bottom-anim-value {:toValue -20})
+                              (anim/timing alpha-value {:toValue  0.6
+                                                        :duration 500})]))}
+    (let [_ (when-not requested-permission (js/setTimeout hide-panel 10))
+          {:keys [dapp-name]} params
+          {:keys [title description icon]} (get model/permissions requested-permission)]
+      [react/view styles/permissions-panel-container
+       [react/animated-view {:style (styles/permissions-panel-background alpha-value)}]
+       [react/animated-view {:style (styles/permissions-panel bottom-anim-value)}
+        [react/view styles/permissions-panel-icons-container
+         (if dapp?
+           [chat-icon.screen/dapp-icon-permission dapp 48]
+           [react/view styles/permissions-panel-dapp-icon-container
+            [react/text {:style styles/permissions-panel-d-label} "Ð"]])
+         [react/view {:margin-left 3 :margin-right 3}
+          [react/view styles/dot]]
+         [react/view {:margin-right 3}
+          [react/view styles/dot]]
+         [react/view styles/permissions-panel-ok-icon-container
+          [icons/icon :icons/ok styles/permissions-panel-ok-ico]]
+         [react/view {:margin-left 3 :margin-right 3}
+          [react/view styles/dot]]
+         [react/view {:margin-right 3}
+          [react/view styles/dot]]
```

Comment:
Maybe some function could be added here for more visibility.
And move styles in `styles/` ns?

Reply:
tbh i agree here with desktop team , i don't see any reason to move all maps to styles, we can keep "small" ones and move only "big" maps, tbh it takes a lot of time to generate style names and move them to styles ns

Reply:
Downside is that it prevents reuse. With proper components it shouldn't be a pb. We have way too much similar styles everywhere.

But why not? Should we discuss that with others?

Reply:
let's discuss for sure , but i believe @vkjr discussed couple of times, and tbh in 99% styles used only once